includes/discovery/discovery-protocols.inc.php

```diff
@@ -310,6 +310,21 @@ function ($v) {
                 foreach ($value as $lldpV2RemLocalDestMACAddress => $value) {
                     foreach ($value as $lldpV2RemIndex => $lldpv2_array_entries) {
                         foreach ($lldpv2_array_entries as $key => $value) {
+                            // Check if 'lldpV2RemPortIdSubtype' is of type 'interfaceName(5)' and create the variable $rem_port_id_subtype to use it later
+                            if ($key === 'lldpV2RemPortIdSubtype' && $value == '5') {
+                                $rem_port_id_subtype = '5';  // Assign the port subtype
```

Comment:
This variable is never reset. So after the first set to '5', all lldpVeRemPortId will be converted. I would go another route : just import lldpV2RemPortIdSubtype in the $lldp_array, and do the conversion further down, around line 340.
And to convert the hex to a string, I would go with this function which does exactly that: https://github.com/librenms/librenms/blob/d23ecf551d44c0b8585f82c85c38e9b094c49200/LibreNMS/Util/StringHelpers.php#L178
Can be called like so : 
`use LibreNMS\Util\StringHelpers;` at the beginning of the file 
and
`StringHelpers::hexToAscii($par, ':')`